src/state_machine/coordinator/fire.rs

```diff
@@ -608,7 +623,13 @@ impl<Aggregator: AggregatorTrait> Coordinator<Aggregator> {
                             // bad_shares is a set of signer_ids
                             for bad_signer_id in bad_shares {
                                 // verify public shares are bad
-                                let dkg_public_shares = &self.dkg_public_shares[bad_signer_id];
+                                let Some(dkg_public_shares) =
+                                    self.dkg_public_shares.get(bad_signer_id)
+                                else {
+                                    warn!("Signer {signer_id} reported BadPublicShares from invalid signer_id {bad_signer_id}, mark {signer_id} as malicious");
+                                    self.malicious_dkg_signer_ids.insert(*signer_id);
```

Comment:
Are we sure that `signer_id` is malicious in this case? I think so, because that means that `signer_id` sent an ID for a signer that doesn't exist, but I'm not entirely sure that this is the case, and that this is the only reason why an entry would be missing from `dkg_public_shares`.

Reply:
In a `FIRE` coordinator world, the coordinator specifies which signers are taking part in successive stages of `DKG`.  So the only way I can see us getting here is if a signer ignores the coordinator and reports a signer who isn't taking part as sending bad shares.  Unless the signer code is broken somehow that seems like a malicious act to me.

Of course, a coordinator can also be malicious.  But then it doesn't need this code block to maliciously report a signer as malicious.

Reply:
Hmmm, I see.

Do we have a test for running DKG using the FIRE coordinator where a malicious signers is detected, excluded, and then DKG succeeds during the second run? Definitely not for this PR though. Also, I don't think that this behavior is necessary for sBTC yet, since we still need to figure out what we're going to do when we detect a misbehaving signer.

Reply:
`malicious_signers_dkg` detects, but does not exclude and continue.

Reply:
`DkgPublicShares` are sent to everyone @matteojug, they aren't encrypted for specific users like `DkgPrivateShares` are.  Unless I'm missing something about how `libp2p` works?

Reply:
I made an ticket to enhance the protocol to address commitment inconsistencies: https://github.com/stacks-sbtc/wsts/issues/190

This is orthogonal to this PR, so what do you think we should we do here @matteojug?  I only changed this particular chunk of code to remove unchecked map access, which is an ongoing concern for @djordon (and to be clear, I fully support removing such code).

Reply:
Looking at the specific code again, I think commitment inconsistency isn't the issue here: while this is a signer reporting bad public shares, the actual error is that the reporter is referencing a signer who isn't taking part in this round of DKG.  This isn't a question of consistency, since the coordinator is the one who declares who's still taking part.

So I think we should leave this code as is in this PR.